src/libsync/propagateremotedeleteencrypted.cpp

```diff
@@ -70,48 +84,149 @@ void PropagateRemoteDeleteEncrypted::slotFolderEncryptedMetadataReceived(const Q
 {
     if (statusCode == 404) {
         qCDebug(PROPAGATE_REMOVE_ENCRYPTED) << "Metadata not found, ignoring.";
+        if (_isRootFolder) {
+            deleteNestedRemoteItemsFinished();
+            return;
+        }
         emit finished(true);
         return;
     }
 
-    qCDebug(PROPAGATE_REMOVE_ENCRYPTED) << "Metadata Received, Preparing it for the new file.";
-
-    // Encrypt File!
     FolderMetadata metadata(_propagator->account(), json.toJson(QJsonDocument::Compact), statusCode);
 
-    QFileInfo info(_propagator->fullLocalPath(_item->_file));
-    const QString fileName = info.fileName();
-
-    // Find existing metadata for this file
-    bool found = false;
-    const QVector<EncryptedFile> files = metadata.files();
-    for (const EncryptedFile &file : files) {
-        if (file.originalFilename == fileName) {
-            metadata.removeEncryptedFile(file);
-            found = true;
-            break;
+    qCDebug(PROPAGATE_REMOVE_ENCRYPTED) << "Metadata Received, Preparing it for the new file(s).";
+
+    if (_isRootFolder) {
+        qCDebug(PROPAGATE_REMOVE_ENCRYPTED) << "It's a root encrypted folder. Let's remove nested items first.";
+
+        bool isFolderEmpty = true;
+
+        _propagator->_journal->listFilesInPath(_item->_file.toUtf8(), [&isFolderEmpty, this](const OCC::SyncJournalFileRecord &record) {
+            _nestedItems[record._e2eMangledName]= record._path;
+            isFolderEmpty = false;
+        },
+        [&metadata, this]() {
+            if (_nestedItems.size() > 0) {
+                for (auto it = _nestedItems.constBegin(); it != _nestedItems.constEnd(); ++it) {
+                    const QString fileName = QFileInfo(it.value()).fileName();
+
+                    // remove the metadata for the current item if present
+                    const QVector<EncryptedFile> files = metadata.files();
+                    for (const EncryptedFile &file : files) {
+                        if (file.originalFilename == fileName) {
+                            metadata.removeEncryptedFile(file);
+                            break;
+                        }
+                    }
+                }
+            }
+        });
+
+        if (isFolderEmpty) {
+            deleteNestedRemoteItemsFinished();
+            return;
         }
-    }
+    } else {
+        // Encrypt File!
+        QFileInfo info(_propagator->fullLocalPath(_item->_file));
+        const QString fileName = info.fileName();
 
-    if (!found) {
-        // The removed file was not in the JSON so nothing else to do
-        emit finished(true);
-        return;
+        // Find existing metadata for this file
+        bool found = false;
+        const QVector<EncryptedFile> files = metadata.files();
```

Comment:
@er-vin Well, in fact, I keep changing this part, and early morning I've realized I can just create empty metadata and push it in UPDATE since we are removing all the files anyway, so the whole lambda part could be gone completely.